doc/migrations/v0.43-v0.44.md

```diff
@@ -0,0 +1,117 @@
+# Migrating to libp2p@44 <!-- omit in toc -->
+
+A migration guide for refactoring your application code from libp2p v0.43.x to v0.44.0.
+
+## Table of Contents <!-- omit in toc -->
+
+- [Connection Manager](#connection-manager)
+- [Connection Gater](#connection-gater)
+
+## Connection Manager
+
+The connection manager has been refactored to be simpler and faster.  The changes
+are mostly internal but some configuration options have changed.
+
+**Before**
+
+```js
+import { createLibp2p } from 'libp2p'
+
+const node = await createLibp2p({
+  connectionManager: {
+    // this option has been removed as auto dial now runs in response to peers
+    // disconnecting and is no longer on a timer
+    autoDialInterval: 0,
+
+    // this has been renamed to `maxPeerAddrsToDial`
+    maxAddrsToDial: 20,
+
+    // this has been renamed to `maxParallelDialsPerPeer`
+    maxDialsPerPeer: 20,
+
+    // these should now be ranges expressed as MultiaddrFilters
+    allow: [
+      '/ip4/0.0.0.0/tcp/123'
+    ],
+
+    // these should now be ranges expressed as MultiaddrFilters
+    deny: [
+      '/ip4/0.0.0.0/tcp/123'
+    ]
+  }
+})
+```
+
+**Before**
+
+```js
+import { createLibp2p } from 'libp2p'
+import { MultiaddrFilter } from from '@multiformats/multiaddr'
+
+const node = await createLibp2p({
+  connectionManager: {
+    // how many peers to dial at once while trying to ensure the node
+    // is above minConnections
+    autoDialConcurrency: 25,
+
+    // a low value allows user-initiated dials to take priority over
+    // auto dials
+    autoDialPriority: 0,
+
+    // this was previously named `maxAddrsToDial`
+    maxPeerAddrsToDial: 20,
```

Comment:
PerConnection attempt? Or total(then we banlist the peer as bad??)

Might be good to call out which one and how in jsdoc, or in the name if possible.

Reply:
Did you check [the jsdoc for this config option](https://github.com/libp2p/js-libp2p/blob/fix/refactor-connection-manager/src/connection-manager/index.ts#L79-L83)?  If it's not clear could you suggest an edit?